content/docs/instrumenting/http_sd.md

```diff
@@ -0,0 +1,96 @@
+---
+title: HTTP SD
+sort_rank: 7
+---
+
+# Writing HTTP Service Discovery
+
+Prometheus provides a generic [HTTP Service Discovery](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#http_sd_config),
+that enables it to discover targets over an HTTP endpoint.
+
+The HTTP Service Discovery is complimentary to the supported service
+discovery mechanisms, and is an alternative to [File-based Service Discovery](https://prometheus.io/docs/guides/file-sd/#use-file-based-service-discovery-to-discover-scrape-targets).
+
+## Comparison between File-Based SD and HTTP SD
+
+Here is a table comparing our two generic Service Discovery implementations.
+
+| Item | File SD | HTTP SD |
+| ---- | ------- | ------- |
+| Event Based | Yes, via inotify | No |
+| Update frequency | Instant, thanks to inotify | Following refresh_interval |
+| Format | Yaml or JSON | JSON |
+| Transport | Local file | HTTP/HTTPS |
+| Security | File-Based security | TLS, Basic auth, Authorization header, OAuth2 |
+
+## Requirements of HTTP SD endpoints
+
+If you implement an HTTP SD endpoint, here is a few requirements you should be
+aware of.
+
+The response is consumed as is, unmodified. On each refresh interval (default: 1
+minute), Prometheus will do a GET request to the HTTP SD endpoint. The GET
+request contains a `X-Prometheus-Refresh-Interval-Seconds` HTTP header with the
```

Comment:
Why? What do you suggest implementers do with it?

Reply:
X-Prometheus-Refresh-Interval-Seconds is 1, they might cache on their side. I don't want to call this explicitly, I just want to guarantee that we will keep this field in the future.